commit eca496d8fe4bcad503e61dd4510907478ce00118
Author: Maxime Ripard <maxime.ripard@free-electrons.com>
Date:   Wed Nov 8 16:47:10 2017 +0100

    ASoC: sun8i-codec: Fix left and right channels inversion
    
    commit 18c1bf35c1c09bca05cf70bc984a4764e0b0372b upstream.
    
    Since its introduction, the codec had an inversion of the left and right
    channels. It turned out to be pretty simple as it appears that the codec
    doesn't have the same polarity on the LRCK signal than the I2S block.
    
    Fix this by inverting our bit value for the LRCK inversion.
    
    Fixes: 36c684936fae ("ASoC: Add sun8i digital audio codec")
    Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
    Reviewed-by: Chen-Yu Tsai <wens@csie.org>
    Signed-off-by: Mark Brown <broonie@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/sound/soc/sunxi/sun8i-codec.c b/sound/soc/sunxi/sun8i-codec.c
index 038107baf414..c2ceca485d6a 100644
--- a/sound/soc/sunxi/sun8i-codec.c
+++ b/sound/soc/sunxi/sun8i-codec.c
@@ -199,7 +199,7 @@ static int sun8i_set_fmt(struct snd_soc_dai *dai, unsigned int fmt)
 			   value << SUN8I_AIF1CLK_CTRL_AIF1_BCLK_INV);
 	regmap_update_bits(scodec->regmap, SUN8I_AIF1CLK_CTRL,
 			   BIT(SUN8I_AIF1CLK_CTRL_AIF1_LRCK_INV),
-			   value << SUN8I_AIF1CLK_CTRL_AIF1_LRCK_INV);
+			   !value << SUN8I_AIF1CLK_CTRL_AIF1_LRCK_INV);
 
 	/* DAI format */
 	switch (fmt & SND_SOC_DAIFMT_FORMAT_MASK) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1096_eca496d8fe4bcad503e61dd4510907478ce00118.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1096_eca496d8fe4bcad503e61dd4510907478ce00118.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
